src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,992 @@
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        [Fact]
+        public void AddDirectory_WithHashAlgorithm()
+        {
+            var alg = "keccak-512";
+            var options = new AddFileOptions {Hash = alg};
+            var temp = MakeTemp();
+            try
+            {
+                var dir = ipfs.UnixFsApi.AddDirectoryAsync(temp, false, options).Result;
+                Assert.True(dir.IsDirectory);
+                Assert.Equal(alg, dir.Id.Hash.Algorithm.Name);
+
+                foreach (var link in dir.Links)
+                {
+                    Assert.Equal(alg, link.Id.Hash.Algorithm.Name);
+                }
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        [Fact]
+        public void AddDirectory_WithCidEncoding()
+        {
+            var encoding = "base32z";
+            var options = new AddFileOptions {Encoding = encoding};
+            var temp = MakeTemp();
+            try
+            {
+                var dir = ipfs.UnixFsApi.AddDirectoryAsync(temp, false, options).Result;
+                Assert.True(dir.IsDirectory);
+                Assert.Equal(encoding, dir.Id.Encoding);
+
+                foreach (var link in dir.Links)
+                {
+                    Assert.Equal(encoding, link.Id.Encoding);
+                }
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        [Fact]
+        public async Task AddDirectoryRecursive_ObjectLinks()
+        {
+            var temp = MakeTemp();
+            try
+            {
+                var dir = await ipfs.UnixFsApi.AddDirectoryAsync(temp, true);
+                Assert.True(dir.IsDirectory);
+
+                var cid = dir.Id;
+                var i = 0;
+                var allLinks = new List<IMerkleLink>();
+                while (cid != null)
+                {
+                    var links = await ipfs.ObjectApi.LinksAsync(cid);
+                    allLinks.AddRange(links);
+                    cid = (i < allLinks.Count) ? allLinks[i++].Id : null;
+                }
+
+                Assert.Equal(6, allLinks.Count);
+                Assert.Equal("alpha.txt", allLinks[0].Name);
+                Assert.Equal("beta.txt", allLinks[1].Name);
+                Assert.Equal("x", allLinks[2].Name);
+                Assert.Equal("x.txt", allLinks[3].Name);
+                Assert.Equal("y", allLinks[4].Name);
+                Assert.Equal("y.txt", allLinks[5].Name);
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        // [Fact]
+        // [Ignore("https://github.com/richardschneider/net-ipfs-engine/issues/74")]
+        // public async Task ReadTextFromNetwork()
+        // {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)